tensorflow/compiler/xla/service/gpu/tests/reduction_vectorization_test.cc

```diff
@@ -113,7 +113,18 @@ ENTRY %fused_computation.371 (param_0: f32[6400,4,8,32]) -> f32[6400,4,8] {
 }
 )";
 
-  const char* expected_optimized_llvm_ir = R"(
+  const char* expected_optimized_llvm_ir = is_built_with_rocm_ ? R"(
```

Comment:
Is it possible to avoid duplication by extracting different substrings (workitem vs. tid.x() and bpermute vs. shfl.sync.down), making them different, and pasting them into the larger string?

Reply:
Done.  PTAL.

Thanks!

Reply:
i actually prefer the duplication because it makes the ROCm and CUDA cases more readable.